src/gl/gl.cpp

```diff
@@ -246,6 +246,12 @@ void gl_draw_indexed_primitive(uint32_t primitivetype, uint32_t vertextype, stru
 
 	newRenderer.setInterpolationQualifier(current_state.shademode ? RendererInterpolationQualifier::SMOOTH : RendererInterpolationQualifier::FLAT);
 
+	if (vertextype != TLVERTEX)
+	{
+		newRenderer.setD3DProjection(&current_state.d3dprojection_matrix);
+		newRenderer.setD3DViweport(&d3dviewport_matrix);
+	}
+
```

Comment:
May I ask why did you move this block? Is this really required?

Reply:
I moved this code to guarantee that the projection and viewport matrices are set before drawing the field shadows. Until now I was drawing other 3d models such as the characters before the field shadows so these matrices were set before drawing the shadows. However, with the changes of this PR, the field shadows (walkmesh) are the first 3d object to be drawn in the frame. By moving this block of code before the draw call deferring I was able to guarantee that these matrices are set before any drawing. Without this change the matrices used when drawing the field shadows have a one frame delay.

As far as I know these matrices are common for all 3d objects so it should be enough to set once at the beginning of the frame. However, I did it this way because it was the method with the less amount of changes required. If this change is not desired I could work around it by adding a new method that sets the matrices for the first 3d object before any drawing. Let me know what you think and thanks for the review as always!

Reply:
To be honest the only part that concerns me is that this code is in common with FF8 which has a tons of tricks in place put here and there to guarantee perfect rendering. This part of the code is one of those and I have no idea if this code change might have an impact ( as you notice there's an if case before this block originally which might return and set basically nothing. Moving this block before that if, now might bring some implications i might not be aware.

If you can find a way to set those in your lighting code class I would appreciate. Thank you!

Reply:
I implemented an alternative way to set the matrices using a new method that is only called when lighting is enabled.

Reply:
Amazing, thank you :)